cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -155,7 +155,8 @@ function(arrow_install_cmake_package PACKAGE_NAME EXPORT_NAME)
   set(CONFIG_CMAKE "${PACKAGE_NAME}Config.cmake")
   set(BUILT_CONFIG_CMAKE "${CMAKE_CURRENT_BINARY_DIR}/${CONFIG_CMAKE}")
   configure_package_config_file("${CONFIG_CMAKE}.in" "${BUILT_CONFIG_CMAKE}"
-                                INSTALL_DESTINATION "${ARROW_CMAKE_DIR}/${PACKAGE_NAME}")
+                                INSTALL_DESTINATION "${ARROW_CMAKE_DIR}/${PACKAGE_NAME}"
+                                NO_CHECK_REQUIRED_COMPONENTS_MACRO)
```

Comment:
@kou I am not sure if the solution to this issue should be to not add the required components macro or fix CMake to force finding the required components before. Let  me know what do you think

Reply:
It does seem the Appveyor failure is related:
```
-- Detecting CXX compile features - done
CMake Error at C:/Miniconda38-x64/envs/arrow/Library/lib/cmake/Arrow/ArrowConfig.cmake:178 (check_required_components):
  Unknown CMake command "check_required_components".
Call Stack (most recent call first):
  CMakeLists.txt:63 (find_package)
```